lib/facter/resolvers/linux/hostname.rb

```diff
@@ -76,6 +81,9 @@ def retrieve_fqdn_for_host_with_ffi(host)
             fqdn = Facter::Util::Resolvers::Ffi::Hostname.getffiaddrinfo(host)
             log.debug("FFI getaddrinfo was called and it retrieved: #{fqdn}")
             fqdn
+          rescue LoadError => e
+            log.debug(e.message)
+            nil
```

Comment:
In the future we could think about creating a feature check for this, like Puppet does with `Puppet.features.something?`, so we won't have to rescue in every place we try to use ffi.